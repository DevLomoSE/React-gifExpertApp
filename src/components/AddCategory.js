import React, { useState } from "react";
import PropTypes from "prop-types";

/**
 * Component to handle the functionality to add new categories.
 *
 * @param setCategories function to mutate the categories state from parent
 */
export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = ( e ) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if(inputValue.trim().length > 0){
            setCategories((cats) => [...cats, inputValue]);
            setInputValue("");
        }
    }

  return (
    <>
      <h2>AddCategory</h2>
      <form onSubmit={handleSubmit}>
        <input type="text" value={inputValue} onChange={handleInputChange} />
      </form>
    </>
  );
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}