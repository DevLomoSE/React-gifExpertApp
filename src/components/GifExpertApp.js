
import React, { useState } from 'react';

export const GifExpertApp = () => {

  const [categories, setCategories] = useState(['One punch', 'Samurai X', 'Yuhi-oh']);

  const handleAdd = (category) => {
    category = 'skdf';
    setCategories( cats => [...cats, category]);
  };

  return (
  <>
    <h2>GifExpertApp</h2>
    <hr />

    <button onClick={ handleAdd }>Add category</button>

    <ol>
      {categories.map( (cat, index) => {
        return <li key={index+1}>{cat}</li>;
      } )}
    </ol>
  </>
  );
};

export default GifExpertApp;
