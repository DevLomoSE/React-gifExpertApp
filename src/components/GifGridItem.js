import React from 'react';

const GifGridItem = ({id, title, img}) => {

  return (
    <div className='card'>
        <img src={img} alt={title}></img>
        <p>{title}</p>
    </div>
  );
};

export default GifGridItem;
