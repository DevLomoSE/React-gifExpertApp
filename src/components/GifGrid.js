import React, { useEffect, useState } from "react";
import GifGridItem from "./GifGridItem";

/**
 * Component to retrieve images about the category.
 *
 * @param category search criteria
 */
const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);

  useEffect(() => {
    getGifs();
  }, []);

  const getGifs = async () => {
    const url = `${process.env.REACT_APP_GIPHY_GIFS_LINK}/search?q=${category}&limit=10&api_key=${process.env.REACT_APP_GIPHY_API_KEY}`;

    const resp = await fetch(url);
    const { data } = await resp.json();

    const gifs = data.map((img) => {
      return {
        id: img.id,
        title: img.title,
        img: img.images?.downsized_medium.url,
      };
    });

    setImages(gifs);
  };

  return (
    <>
      <h3>{category}</h3>
      {images.map( img => {
        return (
            <>
                <GifGridItem 
                    key={img.id}
                    {...img}
                />  
            </>
        );
      })}
    </>
  );
};

export default GifGrid;
