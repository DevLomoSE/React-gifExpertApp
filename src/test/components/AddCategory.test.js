import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";
import "@testing-library/jest-dom";

describe("Testing for <AddCategory />", () => {
  const setCategories = () => {};
  test("should render", () => {
    const wrapper = shallow(<AddCategory setCategories={setCategories} />);
    
    expect(wrapper).toMatchSnapshot();
  });
});
