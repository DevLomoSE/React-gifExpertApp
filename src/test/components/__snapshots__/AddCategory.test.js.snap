// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing for <AddCategory /> should render 1`] = `
<Fragment>
  <h2>
    AddCategory
  </h2>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
